packages/protocol/contracts/layer1/preconf/iface/ILookaheadStore.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@eth-fabric/urc/ISlasher.sol";
+
+/// @title ILookaheadStore
+/// @custom:security-contact security@taiko.xyz
+interface ILookaheadStore {
+    // An array of `LookaheadPayload` will be byte-encoded to be the payload of the
+    // lookahead commitment.
+    struct LookaheadPayload {
+        // Timestamp of the L1 slot
+        uint256 slotTimestamp;
+        // Registration root of the operator in the URC
+        bytes32 registrationRoot;
+        // Index of the Operator's registration merkle tree leaf that contains the validator
+        // for the slot
+        uint256 validatorLeafIndex;
+    }
+
+    struct LookaheadSlot {
+        // Timestamp of the slot.
+        uint256 timestamp;
+        // The preconfer operator's committer address that is fetched from the slashing commitment.
+        address committer;
+        // URC registration root of the operator
+        bytes32 operatorRegistrationRoot;
+        // Index of the Operator's registration merkle tree leaf that contains the validator for the
+        // slot.
+        uint256 validatorLeafIndex;
+    }
+
+    struct LookaheadHash {
+        // The timestamp of the epoch.
+        uint256 epochTimestamp;
+        // Keccak hash of the lookahead slots for the epoch.
+        bytes32 lookaheadHash;
+    }
+
+    struct Config {
+        // The size of the lookahead buffer.
+        uint16 lookaheadBufferSize;
+        // The minimum collateral for a registered operator to post the lookahead.
+        uint80 minCollateralForPosting;
+        // The minimum collateral for a registered operator to preconf.
+        uint80 minCollateralForPreconfing;
+    }
+
+    error CommittmentSignerMismatch();
+    error InvalidLookaheadEpoch();
+    error InvalidSlotTimestamp();
+    error InvalidValidatorLeafIndex();
+    error LookaheadNotRequired();
+    error LookaheadHashNotFound();
+    error OperatorHasBeenSlashed();
+    error OperatorHasInsufficientCollateral();
+    error OperatorHasNotOptedIntoPreconfSlasher();
+    error OperatorHasUnregistered();
+    error PosterHasBeenSlashed();
+    error PosterHasInsufficientCollateral();
+    error PosterHasNotOptedIn();
+    error PosterHasUnregistered();
+    error SlasherIsNotGuardian();
+    error SlotTimestampIsNotIncrementing();
+
+    event LookaheadHashUpdated(uint256 epochTimestamp, bytes32 lookaheadHash);
+    event LookaheadPosted(uint256 indexed timestamp, LookaheadSlot[] lookaheadSlot);
+    event LookaheadPostedByGuardian(uint256 indexed timestamp, LookaheadSlot[] lookaheadSlot);
```

Comment:
how about we reuse `LookaheadHashUpdated` and `LookaheadPosted` by adding a "isGuardian" field to these two events?